nntrainer/optimizers/optimizer_factory.h

```diff
@@ -14,21 +14,35 @@
 #define __OPTIMIZER_FACTORY_H__
 #ifdef __cplusplus
 
+#include <optimizer.h>
 #include <optimizer_internal.h>
 
 namespace nntrainer {
 
+using OptType = ml::train::OptimizerType;
+
 /**
  * @brief Factory creator with copy constructor
  */
 std::unique_ptr<Optimizer> createOptimizer(const std::string &type,
                                            const Optimizer &opt);
 
+/**
+ * @brief Factory creator with copy constructor using enum(integer)
+ */
+std::unique_ptr<Optimizer> createOptimizer(const OptType &type,
+                                           const Optimizer &opt);
+
 /**
  * @brief Factory creator with constructor
  */
 std::unique_ptr<Optimizer> createOptimizer(const std::string &type);
 
+/**
+ * @brief Factory creator with constructor using enum(integer)
+ */
+std::unique_ptr<Optimizer> createOptimizer(const OptType &type);
```

Comment:
Can you also bring back the corresponding function of ccapi which got deleted.

Reply:
Yup, will do in 30 min

Reply:
Done